queue_job/jobrunner/runner.py

```diff
@@ -278,12 +278,28 @@ def _has_queue_job(self):
                 "SELECT 1 FROM pg_tables WHERE tablename=%s", ("ir_module_module",)
             )
             if not cr.fetchone():
+                _logger.debug("%s doesn't seem to be an odoo db", self.db_name)
                 return False
             cr.execute(
                 "SELECT 1 FROM ir_module_module WHERE name=%s AND state=%s",
                 ("queue_job", "installed"),
             )
-            return cr.fetchone()
+            if not cr.fetchone():
+                _logger.debug("queue_job is not installed for db %s", self.db_name)
+                return False
+            cr.execute(
+                """SELECT COUNT(1)
+                FROM information_schema.triggers
+                WHERE event_object_table = %s
+                AND trigger_name = %s""",
+                ("queue_job", "queue_job_notify"),
+            )
+            if cr.fetchone()[0] != 3:  # INSERT, DELETE, UPDATE
+                _logger.debug(
```

Comment:
Since having the module installed without the trigger is never normal, should we log with warn or error level?

Reply:
Good catch.... IMO we should log with error level.

Reply:
makes sense! fixed